skbio/stats/tests/test_composition.py

```diff
@@ -392,6 +393,83 @@ def test_ilr_inv_basis_one_dimension_error(self):
         with self.assertRaises(ValueError):
             ilr_inv(table, basis=basis)
 
+    def test_alr(self):
+        # 2d-composition
+        comp1 = closure(self.cdata1)
+        alr2d_byhand = np.array([np.log(comp1[:, 0]/comp1[:, 1]),
+                                 np.log(comp1[:, 2]/comp1[:, 1])]).T
+        alr2d_method = alr(comp1, denominator_idx=1)
+        npt.assert_allclose(alr2d_byhand, alr2d_method)
+
+        # 1d-composition
+        comp2 = closure(self.cdata2)
+        alr1d_byhand = np.array([np.log(comp2[0]/comp2[1]),
+                                 np.log(comp2[2]/comp2[1])]).T
+        alr1d_method = alr(comp2, denominator_idx=1)
+        npt.assert_allclose(alr1d_byhand, alr1d_method)
```

Comment:
It would be nice to add a test with a 3D array, just to make sure that the ValueError is raised.  I'd take a look at the [test_clr](https://github.com/essicolo/scikit-bio/blob/e25704ec5789f3bf27219e08e71a5bbb52da0904/skbio/stats/tests/test_composition.py#L249) method for an example of this.